File: utilslibhttp/src/main/java/com/tson/utils/lib/http/base/ApiStatus.java
package com.tson.utils.lib.http.base;

/**
 * Created tangxin
 * Time 2019/2/18 7:20 PM
 */
public abstract class ApiStatus<T> implements BaseStatus<T> {

    private static final String TAG = ApiStatus.class.getName();

    @Override
    public void request(String params) {
//        LogUtils.d(TAG, "JDOPushApi params【" + params + "】");
    }

    @Override
    public void before() {
    }

    @Override
    public void error(Throwable throwable) {
    }

    @Override
    public void tokenInvalid(Throwable throwable) {
    }

    @Override
    public void isEmpty() {
    }
}

File: easyBase/src/main/java/com/easy/base/api/ApiStatus.kt
package com.easy.base.api

/**
 * Date 2019/6/17 3:47 PM
 *
 * @author tangxin
 */
abstract class ApiStatus<T> : BaseApiStatus<T> {

    override fun before() {}

    override fun isEmpty() {}

    override fun loadMore(t: T, isRefresh: Boolean) {}

    override fun error(throwable: Throwable) {
        println(throwable.message)
    }

}
